### PR TITLE
CentralPackageVersion dictionary should be ignore case

### LINKED ISSUE #13871: PackageReference and PackageVersion specified with different casing leads to issues in static graph restore

### PR DESCRIPTION
**Pull request by nkolev92**
# Bug

Fixes:  https://github.com/NuGet/Home/issues/13871

## Description

https://github.com/NuGet/NuGet.Client/commit/15691b7351cf0aa2c0e7f613648fe46a3ccb18da regressed the merging of Central Package Version and PackageReference and unfortunately we didn't have proper testing.

This is blocking the current dev insertion

## PR Checklist

- [x] Meaningful title, helpful description and a linked NuGet/Home issue
- [x] Added tests
- [x] ~Link to an issue or pull request to update docs if this PR changes settings, environment variables, new feature, etc.~

### CODE REVIEW

**Review by jeffkl — DISMISSED** [review]

**Review by zivkan — DISMISSED** [review]

**Review by Nigusu-Allehu — DISMISSED** [review]

File: test/NuGet.Core.FuncTests/Msbuild.Integration.Test/MsbuildRestoreTaskTests.cs [hunk 1]
```diff
@@ -1908,5 +1907,51 @@ public async Task MsbuildRestore_WithMissingCPMVersions_RaisesNU1008(bool useSta
...
+            File.WriteAllText(directoryPackagesPropsPath, directoryPackagesProps);
+
+            // Act
+            CommandRunnerResult result = _msbuildFixture.RunMsBuild(pathContext.WorkingDirectory, $"/t:restore /p:RestoreUseStaticGraphEvaluation={useStaticGraphRestore} {projectA.ProjectPath}", ignoreExitCode: true, testOutputHelper: _testOutputHelper);
+
+            // Assert
+            result.Success.Should().BeTrue(because: result.AllOutput);
+            projectA.AssetsFile.LogMessages.Should().HaveCount(0);
+            projectA.AssetsFile.Targets[0].Libraries.Should().HaveCount(1);
+            projectA.AssetsFile.Targets[0].Libraries.Should().HaveCount(1);
```
Comment (donnie-msft) [inline test/NuGet.Core.FuncTests/Msbuild.Integration.Test/MsbuildRestoreTaskTests.cs]: Super duper asserted?

```suggestion
```

File: test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsSolutionRestoreServiceTests.cs [hunk 2]
```diff
@@ -2694,6 +2694,51 @@ public void ToPackageSpec_WithCPMAndVersionOverride_SetsVersionToVersionOverride
             tfm.Dependencies[1].AutoReferenced.Should().BeFalse();
         }
 
+        [Fact]
+        public void ToPackageSpec_WithCPMAndPackageReferenceWithDifferentCasing_CombinesDataCorrect()
```
Comment (donnie-msft) [inline test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsSolutionRestoreServiceTests.cs]: nit: consistent naming
```suggestion
        public void ToPackageSpec_WithCPMAndPackageReferenceWithDifferentCasing_CombinesCorrectly()
```

File: test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsSolutionRestoreServiceTests.cs [hunk 3]
```diff
@@ -2694,6 +2694,51 @@ public void ToPackageSpec_WithCPMAndVersionOverride_SetsVersionToVersionOverride
...
+            // Act
+            var result = VsSolutionRestoreService.ToPackageSpec(projectName, pri);
+
+            // Assert
+            var tfm = result.TargetFrameworks.First();
+
+            tfm.CentralPackageVersions.Should().HaveCount(1);
+            result.RestoreMetadata.CentralPackageVersionsEnabled.Should().BeTrue();
+            tfm.Dependencies.Should().HaveCount(1);
+            tfm.Dependencies[0].Name.Should().Be(packageId);
```
Comment (donnie-msft) [inline test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsSolutionRestoreServiceTests.cs:2736]: I thought this was checking the case of packageId, but I tested this and actually this was already correct before this PR ("foo" in both before and after this PR). The key is the following assertion of `VersionCentrallyManaged` being true. Maybe the test name should reflect that.
eg,
`ToPackageSpec_WithCPMAndPackageReferenceWithDifferentCasing_VersionCentrallyManaged()`

Comment (nkolev92) [reply test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsSolutionRestoreServiceTests.cs:2736]: It's the same thing as the others, the data wasn't combined correctly, I think that's a better name. 
It's VersionCentrallyManaged & the actual Version.
Everything else should be true cased.

**Review by jeffkl — APPROVED** [review]

**Pull request merged**